Sources/Tools/MaxComponent/plResponderLink.cpp

```diff
@@ -1606,10 +1606,10 @@ const TCHAR* plResponderCmdFootSurface::GetInstanceName(IParamBlock2 *pb)
 
 #ifdef UNICODE
     _snwprintf(name, std::size(name), L"Foot Surface (%S)",
-        plArmatureEffectsMgr::SurfaceStrings[pb->GetInt(ParamID(kSurface))]);
+        plArmatureEffectsMgr::SurfaceStrings[pb->GetInt(ParamID(kSurface))].c_str());
 #else
     _snprintf(name, std::size(name), "Foot Surface (%s)",
-        plArmatureEffectsMgr::SurfaceStrings[pb->GetInt(ParamID(kSurface))]);
+        plArmatureEffectsMgr::SurfaceStrings[pb->GetInt(ParamID(kSurface))].c_str());
 #endif
```

Comment:
`.c_str()` is iffy here - technically this passes UTF-8 into a format expecting ANSI (in practice it doesn't matter, because the strings are pure ASCII). But this entire block can be simplified to:

```suggestion
    _sntprintf(name, std::size(name), _T("Foot Surface (%s)"),
        ST2T(plArmatureEffectsMgr::GetSurfaceName(pb->GetInt(ParamID(kSurface)))));
```

Reply:
This suggestion won't work correctly - it passes a non-trival type (`ST::string`) into printf, and the wrong formatter is used when `UNICODE` is not defined (on Max 2012 and lower). Needs to be `%S` in that situation.

Reply:
`_sntprintf` maps to the narrow/wide variant depending on whether `UNICODE` is enabled, and `ST2T` does the conversion from `ST::string` to `TCHAR*`, no? (This follows the same pattern as most of the other `GetInstanceName` implementations.)

Reply:
It maps to either `_snprintf` (which requires char* to be `%s`) when `UNICODE` is not defined or to `_snwprintf` (which requires char* to be `%S`) when `UNICODE` is defined. Further, non-trival types, such as `ST::string` cannot be passed as arguments to `printf` in general. We had lots of fun fixing those kinds of crashing bugs when we first changed many  C strings to `plString`.

Reply:
I don't understand... Unless I'm missing something, it should match up in both cases:

* If `UNICODE` is defined, then `ST2T(s)` becomes `s.to_wchar().data()`, which is a `wchar_t*`, which is appropriate for `%s` in `_snwprintf`
* If `UNICODE` is *not* defined, then `ST2T(s)` becomes `s.to_latin_1().data()`, which is a `char*`, which is appropriate for `%s` in `_snprintf`

It shouldn't make any difference that it was an `ST::string` originally, because it's explicitly converted to a plain C string before it reaches the printf. The C string *is* based on a temporary buffer, but that shouldn't be a problem - IIRC, C++ guarantees that temporary intermediate values stay alive until the end of the statement (i. e. during the entire `_sntprintf` call).

Reply:
I think the problem here is that I was forgetting ehat exactly `ST2T` returns. Your suggestion is fine.